cavatica-proxy/src/main/java/io/kidsfirst/keys/cavatica/proxy/CavaticaProxy.java

```diff
@@ -0,0 +1,144 @@
+  private static String[] HTTP_ALLOWED_METHODS = {"GET", "POST", "PUT", "PATCH", "DELETE"};
+
+  @Override
+  public String processEvent(JSONObject event, String userId)
+    throws IllegalArgumentException, URISyntaxException, IOException, ParseException {
+
+    if (userId == null) {
+      throw new IllegalArgumentException("User ID not found.");
+    }
+    String cavaticaKey = getCavaticaKey(userId);
+
+    try {
+
+      JSONParser parser = new JSONParser();
+      JSONObject eventBody = (JSONObject) parser.parse((String) event.get("body"));
+
+      // Path
+      String path = (String) eventBody.get("path");
+
+      // Method - confirm it is in HTTP_ALLOWED_METHODS or throw error
+      String method = ((String) eventBody.get("method")).toUpperCase();
+      if ( Arrays.stream(HTTP_ALLOWED_METHODS).noneMatch(allowed -> allowed.equals(method)) ) {
+        // Invalid method provided
+        throw new IllegalArgumentException("Provided method '"+method+"'is not allowed.");
+      }
+
+      // Body - default to null
+      JSONObject bodyJson = ((JSONObject) eventBody.get("body"));
+      String body = bodyJson == null ? null : bodyJson.toJSONString();
+
+      String response = sendCavaticaRequest(cavaticaKey, path, method, body);
+
+      return response;
+
+    } catch (ClassCastException e) {
+      throw new IllegalArgumentException("Exception thrown accessing request data: " + e.getMessage());
+    }
+
+  }
+
+  private String getCavaticaKey(String userId) throws IllegalArgumentException {
+    List<Secret> allSecrets = SecretDao.getSecret("cavatica", userId);
+
+    if (!allSecrets.isEmpty()) {
+      Secret secret = allSecrets.get(0);
+
+      String secretValue = secret.getSecret();
+
+      return KMSUtils.decrypt(secretValue);
+
+    } else {
+      throw new IllegalArgumentException("No cavatica secret found, unable to send request.");
+    }
+  }
+
+
+  private String sendCavaticaRequest(String cavaticaKey, String path, String method, String body)
+    throws URISyntaxException, IOException {
+
+    URL url = new URL(cavaticaApiRoot+path);
+
+    HttpsURLConnection con = (HttpsURLConnection) url.openConnection();
+    con.setRequestMethod(method);
+
+    // standard connection setup
+    con.setInstanceFollowRedirects(true);
+    con.setConnectTimeout(1000);
+    con.setReadTimeout(5000);
+
+
+    // Add secret key
+    con.setRequestProperty("X-SBG-Auth-Token", cavaticaKey);
+
+    // Add body
+    if(body != null) {
+      con.setRequestProperty(HttpHeaders.CONTENT_TYPE, ContentType.APPLICATION_JSON.getMimeType());
+      con.setDoOutput(true);
+      DataOutputStream out = new DataOutputStream(con.getOutputStream());
+      out.writeBytes(body);
+      out.flush();
+      out.close();
+    }
+
+    int status = con.getResponseCode();
+
+    BufferedReader reader = new BufferedReader(new InputStreamReader(con.getInputStream()));
+    StringBuffer content = new StringBuffer();
```

Comment:
`StringBuffer` is used when you need thread safety as each method call uses synchronize. This means it is fairly slow which could be bad for the billing in lambda. 

Use `StringBuilder` instead.